### PR TITLE
added lottie support

### PR DESCRIPTION
**Pull request by kavi-ivykids**

### CODE REVIEW

File: lib/ui/components/styled_components/styled_lottie.widget.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:lottie/lottie.dart';
+import 'package:yc_app_utils/models/styled_component/styled_lottie.model.dart';
+
+class StyledLottieWidget extends StatelessWidget {
+  final StyledLottieModel styledLottieModel;
+  const StyledLottieWidget({required this.styledLottieModel, super.key});
```
Comment (naman-jn) [inline lib/ui/components/styled_components/styled_lottie.widget.dart:7]:
```suggestion
  const StyledLottieWidget({required this.styledLottieModel, super.key,});
```

Comment (naman-jn) [inline lib/ui/components/styled_components/styled_lottie.widget.dart:7]:
```suggestion
  final StyledLottieModel styledLottieModel;
  
  const StyledLottieWidget({required this.styledLottieModel, super.key});
```

File: lib/models/styled_component/styled_lottie.model.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      'url': url,
+      'width': width,
+      'height': height,
+    };
+  }
+
+  factory StyledLottieModel.fromMap(Map<String, dynamic> map) {
+    return StyledLottieModel(
+      url: map['url'] as String,
+      width: map['width'] != null ? map['width'] as int : null,
```
Comment (naman-jn) [inline lib/models/styled_component/styled_lottie.model.dart:26]:
```suggestion
      width: map['width'] != null ?  int.tryParse(map['width']?.toString()) : null,
```

**Review by naman-jn — CHANGES_REQUESTED** [review]

File: lib/ui/components/styled_components/styled_lottie.widget.dart (hunk 1 above)
Comment (kavi-ivykids) [reply lib/ui/components/styled_components/styled_lottie.widget.dart:7]: added

File: lib/models/styled_component/styled_lottie.model.dart (hunk 2 above)
Comment (kavi-ivykids) [reply lib/models/styled_component/styled_lottie.model.dart:26]: fixed

File: lib/ui/components/styled_components/styled_lottie.widget.dart (hunk 1 above)
Comment (kavi-ivykids) [reply lib/ui/components/styled_components/styled_lottie.widget.dart:7]: fixed

File: lib/models/styled_component/styled_lottie.model.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+
+import 'package:yc_app_utils/models/styled_component/styled_component_union.dart';
+
+class StyledLottieModel with StyledComponentUnion {
+  String url;
+  int? width;
+  int? height;
+  StyledLottieModel({
```
[the remark this answers is not in the document]
Comment (kavi-ivykids) [reply lib/models/styled_component/styled_lottie.model.dart:9]: fixed

**Review by naman-jn — APPROVED** [review]

**Pull request merged**